ruby/lib/ci/queue/file_flaky_supplier.rb

```diff
@@ -0,0 +1,18 @@
+module CI
+  module Queue
+    class FileFlakySupplier
+      FileParseError = Class.new(StandardError)
+
+      def initialize(filepath)
+        @test_list = JSON.parse(::File.read(filepath))
+        unless @test_list.is_a?(Array)
+          raise FileParseError, 'File must contain a JSON encoded array'
+        end
+      end
+
+      def include?(test)
+        @test_list.include?("#{test.class}##{test.name}")
```

Comment:
This wouldn't work for RSpec. I think this method should either receive a `SingleExample` or a `SingleExample#id`